packages/contracts-bedrock/contracts/L1/SystemConfig.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {
+    OwnableUpgradeable
+} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { Semver } from "../universal/Semver.sol";
+
+/**
+ * @title SystemConfig
+ * @notice This contract is used to update L2 configuration via L1
+ */
+contract SystemConfig is OwnableUpgradeable, Semver {
+    uint256 public constant VERSION = 0;
+
+    uint256 public overhead;
+    uint256 public scalar;
+    address public batcher;
+
+    event ConfigUpdate(
+        uint256 indexed version,
+        UpdateType indexed updateType,
+        bytes data
+    );
+
+    enum UpdateType {
+        BATCHER,
+        GAS_CONFIG
+    }
+
+    constructor(uint256 _overhead, uint256 _scalar, address _batcher) Semver(0, 0, 1) {
+        overhead = _overhead;
+        scalar = _scalar;
+        batcher = _batcher;
```

Comment:
This should also take the owner and call `initialize` so that the impl isn't uninitialized

Reply:
isn't `initialize` called by the proxy / deployment setup? If the constructor calls it, what's the point of having it in a separate function?